api/src/main/java/org/openmrs/ServiceOrder.java

```diff
@@ -0,0 +1,208 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+/**
+ * Provides properties for several order types like TestOrder , ReferralOrder
+ * and others to accommodate referrals depending on the openmrs implementation
+ * use case
+ * 
+ * @since 2.5.0
+ */
+public abstract class ServiceOrder extends Order {
```

Comment:
thanks martin,is this class intended to be used some where,and if so we can also add the code where it is used

Reply:
True ,thanks @HerbertYiga  there's a sub-task following this .If we get it right here . the rest will be history.